src/pretix/base/customersso/oidc.py

```diff
@@ -139,6 +139,17 @@ def oidc_validate_and_complete_config(config):
                 )
             )
 
+    if "query_parameters" in config and config["query_parameters"]:
+        for qp in config["query_parameters"].split("&"):
+            # Very rudimentary check to avoid the most common footguns:
+            # - No ? in the query parameters
+            # - Max of one = (to split key and value)
+            # - One key, one value. Not just keys with no value.
+            if not (qp.count('=') == 1 and qp.count('?') == 0 and len(list(filter(None, qp.split('=')))) == 2):
+                raise ValidationError(
+                    _(f'Query parameter {qp} is invalid.')
+                )
+
```

Comment:
I'd just parse and re-encode it using the library functions -- so the user sees what will be transmitted in the end:
```suggestion
    if "query_parameters" in config and config["query_parameters"]:
         config["query_parameters"] = urllib.parse.urlencode(
             urllib.parse.parse_qsl(config["query_parameters"])
         )
```

Reply:
I was thinking yesterday that there was such a function in urllib, but apparently I was too cooked to actually find it... Thank you for the proposal; that's a lot better!